gufe/chemicalsystem.py

```diff
@@ -126,6 +126,34 @@ def total_charge(self):
                 total_charge += fc
         return total_charge
 
+    def isin(self, item: Component | type[Component]) -> list[Component]:
+        """Check if a Component or Component type is in this ChemicalSystem and return all matches.
+
+        Parameters
+        ----------
+        item : Component or type of Component
+            The Component instance or class to check for.
+
+        Returns
+        -------
+        list[Components]
+            Returns a list of matching Components.
+        """
+        # check the input types
+        if not (isinstance(item, Component) or (isinstance(item, type) and issubclass(item, Component))):
+            raise TypeError("`item` must be an instance or subclass of `Component`")
+
+        matches = []
+        for comp in self._components.values():
+            if isinstance(item, Component):
+                if comp == item:
+                    matches.append(comp)
+            elif issubclass(item, Component):
+                if isinstance(comp, item):
+                    matches.append(comp)
+
+        return matches
```

Comment:
O I thought we were going to keep both the bool return and just matches if we asked for matches. 

If we're keeping matches, would `get` be better than `isin`? The latter usually makes you expect a bool by default.

Reply:
Ahh I thought based on what Alyssa wrote above, we wanted this. Yeah `get` might make more sense based on the return type we now have.